src/commands/deploy.rs

```diff
@@ -505,20 +505,30 @@ impl DeployCommand {
         bindle_version: String,
         app_id: Uuid,
     ) -> Result<Uuid> {
-        let revisions = CloudClient::list_revisions(cloud_client).await?;
-        let revision = revisions
-            .items
-            .iter()
-            .find(|&x| x.revision_number == bindle_version && x.app_id == app_id);
-        Ok(revision
-            .ok_or_else(|| {
-                anyhow!(
-                    "No revision with version {} and app id {}",
-                    bindle_version,
-                    app_id
-                )
-            })?
-            .id)
+        let mut revisions = cloud_client.list_revisions().await?;
+
+        loop {
+            if let Some(revision) = revisions
+                .items
+                .iter()
+                .find(|&x| x.revision_number == bindle_version && x.app_id == app_id)
+            {
+                return Ok(revision.id);
+            }
+
+            if revisions.is_last_page {
+                break;
+            }
+
+            revisions.page_index = revisions.page_index + 1;
```

Comment:
should we probably do `fn list_revision_page(page_index)` instead? changing the previously fetched response does not feel right. But then i am a total noob in Rust, so please ignore me if it does not make sense